sys/net/application_layer/nanocoap/sock.c

```diff
@@ -31,35 +31,39 @@
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-ssize_t nanocoap_request(coap_pkt_t *pkt, sock_udp_ep_t *local, sock_udp_ep_t *remote, size_t len)
+int nanocoap_connect(sock_udp_t *sock, sock_udp_ep_t *local, sock_udp_ep_t *remote)
 {
-    ssize_t res;
-    size_t pdu_len = (pkt->payload - (uint8_t *)pkt->hdr) + pkt->payload_len;
-    uint8_t *buf = (uint8_t*)pkt->hdr;
-    sock_udp_t sock;
-
     if (!remote->port) {
         remote->port = COAP_PORT;
     }
 
-    res = sock_udp_create(&sock, local, remote, 0);
-    if (res < 0) {
-        return res;
-    }
+    return sock_udp_create(sock, local, remote, 0);
+}
+
+ssize_t nanocoap_request(sock_udp_t *sock, coap_pkt_t *pkt, size_t len)
+{
+    ssize_t res = -EAGAIN;
+    size_t pdu_len = (pkt->payload - (uint8_t *)pkt->hdr) + pkt->payload_len;
+    uint8_t *buf = (uint8_t*)pkt->hdr;
+    uint32_t id = coap_get_id(pkt);
 
     /* TODO: timeout random between between ACK_TIMEOUT and (ACK_TIMEOUT *
      * ACK_RANDOM_FACTOR) */
     uint32_t timeout = CONFIG_COAP_ACK_TIMEOUT * US_PER_SEC;
-    unsigned tries_left = CONFIG_COAP_MAX_RETRANSMIT + 1;  /* add 1 for initial transmit */
-    while (tries_left) {
 
-        res = sock_udp_send(&sock, buf, pdu_len, NULL);
-        if (res <= 0) {
-            DEBUG("nanocoap: error sending coap request, %d\n", (int)res);
-            break;
+    /* add 1 for initial transmit */
+    unsigned tries_left = CONFIG_COAP_MAX_RETRANSMIT + 1;
+
+    while (tries_left) {
+        if (res == -EAGAIN) {
```

Comment:
For next squashing round: It would help in reviewing (or hindsight) if the commit message also added a note that the retransmission code is refactored with (what seems to me but I'm not sure right now) equivalent retransmission properties (or was the timeout not doubled originally? If so, that deserves an even clearer note.)

Reply:
Can I do the squashing now?

Reply:
Yes

Reply:
I added a not to 6f161569f469d8640a376ff8f5eb6e7cdcdb37cc